lib/galaxy_test/api/test_workflows.py

```diff
@@ -5444,6 +5444,14 @@ def test_cannot_run_against_other_users_history(self):
         run_workflow_response = self._post(f"workflows/{workflow_id}/invocations", data=workflow_request, json=True)
         self._assert_status_code_is(run_workflow_response, 403)
 
+    def test_cannot_run_workflow_as_anon(self):
+        workflow = self.workflow_populator.load_workflow(name="test_for_run_anon_user")
+        workflow_request, _, workflow_id = self._setup_workflow_run(workflow)
+        with self._different_user(anon=True):
+            run_workflow_response = self._post(f"workflows/{workflow_id}/invocations", data=workflow_request, json=True)
+            self._assert_status_code_is(run_workflow_response, 403)
+            self._assert_error_code_is(run_workflow_response, error_codes.error_codes_by_name["USER_NO_API_KEY"])
```

Comment:
I suppose you want to keep only one of the last 2 lines?

Reply:
We do this in various other parts of this file as well, like:

https://github.com/galaxyproject/galaxy/blob/6e7160bae5b82fea171239b72b756177f725c64b/lib/galaxy_test/api/test_workflows.py#L735-L737

..., so I decided to leave the `self._assert_status_code_is` part in. Should I remove it?

Reply:
Ah, no, you are right, I misread the asserts, sorry!